test/e2e/framework/pod/dial.go

```diff
@@ -91,6 +92,20 @@ func (d *Dialer) DialContainerPort(ctx context.Context, addr Addr) (conn net.Con
 	}
 	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: transport}, "POST", req.URL())
 
+	tunnelingDialer, err := portforward.NewSPDYOverWebsocketDialer(req.URL(), restConfig)
+	if err != nil {
+		return nil, err
+	}
+	// First attempt tunneling (websocket) dialer, then fallback to spdy dialer.
+	dialer = portforward.NewFallbackDialer(tunnelingDialer, dialer, func(err error) bool {
+		if httpstream.IsUpgradeFailure(err) || httpstream.IsHTTPSProxyError(err) {
+			framework.Logf("fallback to secondary dialer from primary dialer err: %v", err)
+			return true
+		}
+		framework.Logf("unexpected error trying to use websockets for portforward: %v", err)
```

Comment:
I am seeing this in tests where the code worked okay:
```
I0506 21:23:56.781188 6341 exec_util.go:201] unexpected error trying to use websockets for pod exec: <nil>
```

I think an if check is missing here.

Reply:
https://github.com/kubernetes/kubernetes/pull/131642/files

Reply:
Actually, it was the `remotecommand.NewFallbackExecutor` output that I was seeing.

We agreed that the assumption that the error is non-nil when these callbacks get invoked is reasonable. Fix for client-go is in https://github.com/kubernetes/kubernetes/pull/131642.